x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/types.ts

```diff
@@ -6,26 +6,35 @@
 
 import * as t from 'io-ts';
 
+export const RuleTypeSchema = t.union([
+  t.literal('query'),
+  t.literal('saved_query'),
+  t.literal('machine_learning'),
+]);
```

Comment:
nit: For small key sizes, io-ts, recommends the keyof with null for speed:
https://github.com/gcanti/io-ts#union-of-string-literals